### ISSUE TITLE
NullReferenceException in version 5.3.10

### ISSUE THREAD

**Issue by zgabi** [nhibernate/nhibernate-core#2937]
With the latest version (5.3.10) of NHibernate I receive a NullReferenceException.
Version 5.3.9 works

.NET 6 RC2

Query:
```c#
DataManager.Query<WeatherDataType>(transaction).FirstOrDefault(t => t.Identification.SensorCode == sensorCode)
```

```c#
[Property(0, TypeType = typeof(WeatherDataIdentificationCompositeUserType))]
[Column(1, Name = nameof(WeatherDataIdentification.DateOfSampling), Unique = false, NotNull = false, Index = "WEATHER_DATEOFSAMPLING")]
[Column(2, Name = nameof(WeatherDataIdentification.InternalId), Unique = false, NotNull = false)]
[Column(3, Name = nameof(WeatherDataIdentification.Position) + "_Latitude", Unique = false, NotNull = false)]
[Column(4, Name = nameof(WeatherDataIdentification.Position) + "_Longitude", Unique = false, NotNull = false)]
[Column(5, Name = nameof(WeatherDataIdentification.SensorCode), Unique = false, NotNull = false)]
[Column(6, Name = nameof(WeatherDataIdentification.SensorId), Unique = false, NotNull = false)]
public virtual WeatherDataIdentification Identification...
```

StackTrace:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at NHibernate.Util.ExpressionsHelper.TryGetMappedNullability(ISessionFactoryImplementor sessionFactory, Expression expression, Boolean& nullable)
   at NHibernate.Linq.Visitors.NullableExpressionDetector.IsNullable(MemberExpression memberExpression, BinaryExpression equalityExpression)
   at NHibernate.Linq.Visitors.NullableExpressionDetector.IsNullable(Expression expression, BinaryExpression equalityExpression)
   at NHibernate.Linq.Visitors.HqlGeneratorExpressionVisitor.TranslateEqualityComparison(BinaryExpression expression)
   at NHibernate.Linq.Visitors.HqlGeneratorExpressionVisitor.VisitExpression(Expression expression)
   at NHibernate.Linq.Visitors.QueryModelVisitor.VisitWhereClause(WhereClause whereClause, QueryModel queryModel, Int32 index)
   at Remotion.Linq.QueryModelVisitorBase.VisitBodyClauses(ObservableCollection`1 bodyClauses, QueryModel queryModel)
   at Remotion.Linq.QueryModelVisitorBase.VisitQueryModel(QueryModel queryModel)
   at NHibernate.Linq.Visitors.QueryModelVisitor.Visit()
   at NHibernate.Linq.Visitors.QueryModelVisitor.GenerateHqlQuery(QueryModel queryModel, VisitorParameters parameters, Boolean root, Nullable`1 rootReturnType)
   at NHibernate.Linq.NhLinqExpression.Translate(ISessionFactoryImplementor sessionFactory, Boolean filter)
   at NHibernate.Hql.Ast.ANTLR.ASTQueryTranslatorFactory.CreateQueryTranslators(IQueryExpression queryExpression, String collectionRole, Boolean shallow, IDictionary`2 filters, ISessionFactoryImplementor factory)
   at NHibernate.Engine.Query.QueryPlanCache.GetHQLQueryPlan(IQueryExpression queryExpression, Boolean shallow, IDictionary`2 enabledFilters)
   at NHibernate.Impl.AbstractSessionImpl.GetHQLQueryPlan(IQueryExpression queryExpression, Boolean shallow)
   at NHibernate.Impl.AbstractSessionImpl.CreateQuery(IQueryExpression queryExpression)
   at NHibernate.Linq.DefaultQueryProvider.PrepareQuery(Expression expression, IQuery& query)
   at NHibernate.Linq.DefaultQueryProvider.Execute(Expression expression)
   at NHibernate.Linq.DefaultQueryProvider.Execute[TResult](Expression expression)
```

**Comment by PleasantD** [comment]
I'm seeing almost this exact error (top 4 levels of the stack trace map exactly).
In our case this happens when trying to perform logical comparisons on a custom type mapped using `ICompositeUserType`.
eg) `query.Where(x => x.Email.Domain == "example.com").ToList()`

**Comment by PleasantD** [comment]
I've tracked this a little further,
It looks like the exception is occurring here:
https://github.com/nhibernate/nhibernate-core/blob/c4174f97bd61115cd2dac0aff3245862852e804f/src/NHibernate/Util/ExpressionsHelper.cs#L166

It turns out that `NHibernate.Type.CompositeCustomType.PropertyNullability` returns `null`, so the indexing on that line fails.

I _think_ what needs to happen there is that if `componentType is CompositeCustomType` then we can "inherit" the nullability from the nullability of the "parent" property. This could be accomplished by removing the last element of the `memberPath` and letting the logic continue with the `entityPersister` property logic.

Can anyone confirm?

**Comment by fredericDelaporte** [comment]
Fixed by #2941. MyGet dev builds for 5.3.11.dev, which should be soon generated, will allow you to check the fix for your case.
